staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -322,6 +324,42 @@ func (b *builder) buildKubeNative(schema *structuralschema.Structural, v2 bool)
 	return ret
 }
 
+func addEmbeddedProperties(s *spec.Schema) {
+	if s == nil {
+		return
+	}
+	for k := range s.Properties {
+		v := s.Properties[k]
+		addEmbeddedProperties(&v)
+		s.Properties[k] = v
+	}
+	if s.Items != nil {
+		addEmbeddedProperties(s.Items.Schema)
+	}
+	if isTrue, ok := s.VendorExtensible.Extensions.GetBool("x-kubernetes-embedded-resource"); ok && isTrue {
+		addTypeMetaProperties(s)
+
+		req := sets.NewString(s.Required...)
+
+		if !req.Has("kind") {
+			s.Required = append(s.Required, "kind")
+		}
+		kindProp := s.Properties["kind"]
+		kindProp.Description = "kind is a string value representing the type of this object. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds"
+		s.Properties["kind"] = kindProp
```

Comment:
not sure if I follow what the above Description assignment does

reading the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190425-structural-openapi.md#unfolding-the-extensions), my understanding

for each of `kind`, `apiVersion` and `metadta`:
- if the embedded schema doesn't define the property, we publish the builtin schema for it
```
  apiVersion:
    type: string
    description: xxxxxx
```
- if the embedded schema defines the property, we publish an analog `allOf` variant logically and'ing the builtin and the defined schema, e.g:
```
  apiVersion:
    allOf:
    // builtin
    - type: string
      description: xxxxxx
    // user-defined
    - pattern: v1
```
through v3->v2 conversion the `allOf` gets dropped entirely. I prefer publishing the builtin schema unconditionally for now (v2)

Reply:
Agree on publishing built in unconditionally for now. This is just tweaking the description to be more accurate for embedded use